src/compiler/scala/tools/nsc/transform/LazyVals.scala

```diff
@@ -54,14 +54,35 @@ abstract class LazyVals extends Transform with TypingTransformers with ast.TreeD
     private val lazyVals = perRunCaches.newMap[Symbol, Int]() withDefaultValue 0
 
     import symtab.Flags._
+    private def flattenThickets(stats: List[Tree]): List[Tree] = stats.flatMap(_ match {
```

Comment:
I've also needed this, but the `stats` weren't necessarily `DefDef`s -- could a universal encoding of a thicket look like, e.g., `Block(stats, EmptyTree)`? Maybe we should call it a shrubbery, though.